ui/src/org/pentaho/di/ui/spoon/ExpandedContentManager.java

```diff
@@ -0,0 +1,192 @@
+   */
+  public static void createExpandedContent( TransGraph parent, String url ) {
+    if ( parent == null ) {
+      return;
+    }
+    Browser browser = getExpandedContentForTransGraph( parent );
+    if ( browser == null ) {
+      browser = new Browser( parent, SWT.NONE );
+    }
+    browser.setUrl( url );
+  }
+
+  /**
+   * showTransformationBrowser
+   * 
+   * Creates and shows the web browser for the active TransGraph
+   */
+  public static void showExpandedContent() {
+    showExpandedContent( Spoon.getInstance().getActiveTransGraph() );
+  }
+
+  /**
+   * showExpandedContent( TransGraph graph )
+   * 
+   * @param graph
+   *          TransGraph to create the web browser for. If the wev browser hasn't been created this will create one.
+   *          Else it will just bring the web browser associated to this TransGraph to the top.
+   */
+  public static void showExpandedContent( TransGraph graph ) {
+    if ( graph == null ) {
+      return;
+    }
+    Browser browser = getExpandedContentForTransGraph( graph );
+    if ( browser == null ) {
+      return;
+    }
+    if ( !isVisible( graph ) ) {
+      maximizeExpandedContent( browser );
+    }
+    browser.moveAbove( null );
+    browser.getParent().layout( true );
+    browser.getParent().redraw();
+  }
+
+  /**
+   * getExpandedContentForTransGraph
+   * 
+   * @param graph
+   *          a TransGraph object that will be interrogated for a web browser
+   * @return a web browser that is associated with the TransGraph or null if it has yet to be created.
+   */
+  public static Browser getExpandedContentForTransGraph( TransGraph graph ) {
+    for ( Control control : graph.getChildren() ) {
+      if ( control instanceof Browser ) {
+        return (Browser) control;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * hideExpandedContent
+   * 
+   * hides the web browser associated with the active TransGraph
+   */
+  public static void hideExpandedContent() {
+    hideExpandedContent( Spoon.getInstance().getActiveTransGraph() );
+  }
+
+  /**
+   * hideExpandedContent( TransGraph graph )
+   * 
+   * @param graph
+   *          the TransGraph whose web browser will be hidden
+   */
+  public static void hideExpandedContent( TransGraph graph ) {
+    Browser browser = getExpandedContentForTransGraph( graph );
+    if ( browser == null ) {
+      return;
+    }
+    SashForm sash = (SashForm) browser.getParent().getParent().getParent().getParent();
+    sash.setWeights( savedSashWeights );
+
+    browser.moveBelow( null );
+    browser.getParent().layout( true );
+    browser.getParent().redraw();
+
+    sash.layout( true );
+    sash.redraw();
+  }
+
+  /**
+   * maximizeBrowser
+   * 
+   * @param browser
+   *          the browser object to maximize. We try to take up as much of the Spoon window as possible.
+   */
+  private static void maximizeExpandedContent( Browser browser ) {
+    SashForm sash = (SashForm) browser.getParent().getParent().getParent().getParent();
```

Comment:
Can we add a getter to spoon to return the SashForm so we don't have to do all these getParent() calls?